nodesharing-lib/src/main/java/com/redhat/jenkins/nodesharing/ExecutorJenkins.java

```diff
@@ -42,9 +42,10 @@
 
     private final @Nonnull URL url;
     private final @Nonnull String name;
+    private final String credentialId;
```

Comment:
Please annotate this with `@CheckForNull` as we know it can be null. And perhaps even turn `""` into `null` so client cade does not have to.